cmake/extensions.cmake

```diff
@@ -344,6 +344,17 @@ function(generate_inc_file_for_gen_target
   add_dependencies(${target} ${gen_target})
 endfunction()
 
+function(get_unique_generated_target_name
+    generated_file  # The generated file
+    generated_target_name # the unique name
+    )
+    string(RANDOM LENGTH 8 random_chars)
+    get_filename_component(basename ${generated_file} NAME)
+    string(REPLACE "." "_" basename ${basename})
+    string(REPLACE "@" "_" basename ${basename})
+    set(generated_target_name "gen_${basename}_${random_chars}" PARENT_SCOPE)
+endfunction()
```

Comment:
Code like this should be upstreamed early to minimize the size of the PR.

https://github.com/zephyrproject-rtos/zephyr/pull/10969